### PR TITLE
example/libtest: add malloc/free to libtest

### PR DESCRIPTION
**Pull request by XuNeo**
## Summary

So we can verify flags like kasan are effective or not.

## Impact

Minor

## Testing

Compile libtest and disassemble library_test to verify kasan is actually used.

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, this PR does not meet the NuttX requirements.**

**Missing Information:**

* **Summary:**
    * The summary is too vague. It doesn't explain:
        * **Why** is the change necessary? What problem does it solve?
        * **What** functional part of the code is being changed? Be specific.
        * **How** does the change work? Provide technical details.
    * Include any relevant issue references.
* **Impact:**
    * **Minor** is not descriptive enough. 
    * Address **all** impact points with specific details (even if the answer is NO). 
* **Testing:**
    * **Insufficient detail:**
        * Specify the **exact** build hosts (OS, CPU, compiler versions) and targets (arch, board, config) used.
        * Provide **complete and meaningful** testing logs. "Compile libtest" and "disassemble library_test" are not informative enough.

**Recommendations:**

1. **Expand the Summary:**  Provide a clear and detailed explanation of the problem, the solution, and how it works.
2. **Address All Impact Points:**  Go through each impact category and provide a specific answer, even if it's "NO." Explain any "YES" answers thoroughly.
3. **Provide Detailed Testing Information:**  List all build hosts and targets used, and include complete and relevant testing logs that demonstrate the change's functionality.

File: examples/libtest/libtest.c [hunk 1]
```diff
@@ -35,5 +36,16 @@
 
 void library_test(void)
 {
+  int *p = malloc(sizeof(int));
```
Comment (pkarashchenko) [inline examples/libtest/libtest.c:39]:
```suggestion
  FAR int *p = malloc(sizeof(int));
```

File: examples/libtest/libtest.c [hunk 2]
```diff
@@ -24,6 +24,7 @@
 
 #include <nuttx/config.h>
```
Comment (pkarashchenko) [inline examples/libtest/libtest.c:25]: Not sure why for we need include `config.h` here as no check of configuration is done in the file

Comment (XuNeo) [reply examples/libtest/libtest.c]: Replaced with <nuttx/compiler.h> for `FAR`.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**